src/System.Linq/tests/ToArrayToListToDictionaryTests.cs

```diff
@@ -0,0 +1,469 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+
+
+namespace System.Linq.Tests
+{
+    public class ToArrayToListToDictionaryTests
+    {
+        #region =============  Helper classes =============
+
+        private class EnumerableCollectionTest<T> : IEnumerable<T>
```

Comment:
Nit, but could you rename these helper classes to end with the thing they are, e.g. TestEnumerable, TestReadOnlyCollection, TestCollection, etc., rather than EnumerableCollectionTest, ReadOnlyCollectionTest, etc.?